custom/common/filters.py

```diff
@@ -20,7 +20,6 @@ def loc_to_json(loc):
                 'location_type': loc.location_type.name,
                 'uuid': loc.location_id,
                 'is_archived': loc.is_archived,
-                'level': loc.level,
```

Comment:
The `level` field no longer contains meaningful information and will eventually be removed. I did a project-wide search and attempted to refactor/remove all references to it. It was unclear if references in custom code are related, so I didn't touch them in this PR:

https://github.com/dimagi/commcare-hq/blob/812de382d68f8c9459295d122e5df2b10c1e176a/custom/common/static/common/ko/report_filter.drilldown_options.js#L77

https://github.com/dimagi/commcare-hq/blob/812de382d68f8c9459295d122e5df2b10c1e176a/custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js#L113-L117

https://github.com/dimagi/commcare-hq/blob/812de382d68f8c9459295d122e5df2b10c1e176a/custom/icds_reports/static/js/directives/download/download.directive.js#L144

Reply:
I think you'll need to remove it from https://github.com/dimagi/commcare-hq/blob/812de382d68f8c9459295d122e5df2b10c1e176a/custom/icds_reports/static/icds_reports/js/spec/download.directive.spec.js#L113-L117 since that's a test spec which will probably fail now that it's been removed from the JSON

Reply:
@snopoke on further inspection the one's you're referring to there appear to be location types which have levels assigned when the hierarchy is initialized in javascript:

https://github.com/dimagi/commcare-hq/blob/b1c08a00cc3676a9160659966f06bc21c301f0a1/custom/icds_reports/static/js/directives/location-filter/location-filter.directive.js#L170-L178

The `LocationType` model does not have a `level` field, so I don't think these need to be changed.

Also, I think this `level` field that was removed was not used since it is not set in the main utility method where locations are converted to json:

https://github.com/dimagi/commcare-hq/blob/b1c08a00cc3676a9160659966f06bc21c301f0a1/corehq/apps/locations/util.py#L46-L53

Any code that would have referenced that `level` field would have been broken sometimes because sometimes results were returned containing `level` (by [`RestrictedLocationDrillDown`](https://github.com/dimagi/commcare-hq/blob/b1c08a00cc3676a9160659966f06bc21c301f0a1/custom/common/filters.py#L58)) and other times `level` is omitted (by [`AsyncLocationFilter`](https://github.com/dimagi/commcare-hq/blob/b1c08a00cc3676a9160659966f06bc21c301f0a1/custom/common/filters.py#L57)).